clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10826,6 +10826,9 @@ def err_conflicting_sycl_kernel_attributes : Error<
   "conflicting attributes applied to a SYCL kernel">;
 def err_conflicting_sycl_function_attributes : Error<
    "%0 attribute conflicts with '%1' attribute">;
+def err_sycl_function_attribute_mismatch : Error<
+  "SYCL kernel without %0 attribute can't call a function with this attribute">;
+def note_sycl_function_attribute_mismatch : Note<"function attribute is here">;
```

Comment:
How about using existing diagnostic https://github.com/intel/llvm/blob/9072d49e16cddba544d2a70d79d2a2870d4fa1ce/clang/include/clang/Basic/DiagnosticSemaKinds.td#L3794 
instead adding new one.

Reply:
This one is more accurate.

Reply:
IMO, the existing one is accurate enough to use it instead adding new one.

Reply:
I agree with @Fznamznon the added note isn't particularly more accurate, and isn't work an additional diagnostic.

Reply:
Term "conflicting" kind of suggests that there are two attributes which conflict. In this case there is just one. What does it conflict with?
I would like ESIMD users to get clear error message about what's wrong.

Reply:
My comment was only about note, I'm fine error that you're adding.

Reply:
OK, makes sense then.